src/main/kotlin/com/exactpro/th2/kafka/client/Config.kt

```diff
@@ -81,24 +82,21 @@ class Config (
      */
     val reconnectBackoffMaxMs: Int = 1000
 ) {
-    val aliasToTopic: Map<String, String>
+    @JsonIgnore
     val maxInactivityPeriod: Duration = maxInactivityPeriodUnit.toMillis(maxInactivityPeriod).toDuration(DurationUnit.MILLISECONDS)
 
-    init {
-        require(topicToAlias.isNotEmpty()) { "No topics was provided. Please, check the configuration" }
-
-        val aliasToTopicMutable = HashMap<String, String>(topicToAlias.size)
-        topicToAlias.forEach { (topic: String, alias: String) ->
-            require(alias.isNotEmpty()) { "Session alias can't be empty. Please, check the configuration for $topic" }
-            require(alias !in aliasToTopicMutable) { "Session alias '$alias' duplicated" }
-            aliasToTopicMutable += alias to topic
-        }
-        aliasToTopic = aliasToTopicMutable
+    @JsonIgnore
+    val kafkaStreamToAlias: Map<KafkaStream, String> = aliasToKafkaStream.map { it.value to it.key }.toMap()
 
+    init {
+        require(aliasToKafkaStream.isNotEmpty()) { "'aliasToKafkaStream' can't be empty" }
+        require(kafkaStreamToAlias.size == aliasToKafkaStream.size) { "Duplicated data stream in 'aliasToKafkaStream'" }
         require(reconnectBackoffMaxMs > 0) { "'reconnectBackoffMaxMs' must be positive. Please, check the configuration. $reconnectBackoffMaxMs" }
         require(reconnectBackoffMs > 0) { "'reconnectBackoffMs' must be positive. Please, check the configuration. $reconnectBackoffMs" }
         require(batchSize > 0) { "'batchSize' must be positive. Please, check the configuration. $batchSize" }
         require(maxInactivityPeriod > 0) { "'maxInactivityPeriod' must be positive. Please, check the configuration. $maxInactivityPeriod" }
         require(timeSpan > 0) { "'timeSpan' must be positive. Please, check the configuration. $timeSpan" }
     }
-}
\ No newline at end of file
+}
+
+data class KafkaStream(val topic: String, val key: String?)
```

Comment:
Probably, key should be null by default. In order to skip it in the configuration
```suggestion
data class KafkaStream(val topic: String, val key: String? = null)
```

Reply:
Also, can one topic have many keys? If it can we probably should allow user to specify only topic for the alias